client/a8c-for-agencies/sections/sites/hooks/use-formatted-sites.ts

```diff
@@ -0,0 +1,250 @@
+				monitor.status = 'disabled';
+			} else if (
+				! monitorStatus &&
+				// This check is needed because monitor_site_status is false by default
+				// and we don't want to show the site down status when the site is first connected and the monitor is enabled
+				INITIAL_UNIX_EPOCH !== site.monitor_last_status_change
+			) {
+				monitor.status = 'failed';
+				monitor.value = translate( 'Site Down' );
+				monitor.error = true;
+			} else {
+				monitor.status = 'success';
+			}
+
+			return monitor as MonitorNode;
+		},
+		[ translate ]
+	);
+};
+
+const useFormatPluginData = () => {
+	const translate = useTranslate();
+
+	return useCallback(
+		( site: Site ): PluginNode => {
+			const pluginUpdates = site.is_atomic
+				? site.awaiting_plugin_updates?.filter(
+						( plugin ) =>
+							! PREINSTALLED_PLUGINS.includes( plugin ) &&
+							! AUTOMOMANAGED_PLUGINS.includes( plugin ) &&
+							! ECOMMERCE_BUNDLED_PLUGINS.includes( plugin ) &&
+							! Object.keys( PREINSTALLED_PREMIUM_PLUGINS ).includes( plugin )
+				  )
+				: site.awaiting_plugin_updates;
+
+			if ( ! pluginUpdates ) {
+				return {
+					value: '',
+					status: 'disabled',
+					type: 'plugin',
+					updates: 0,
+				};
+			}
+			return {
+				value: `${ pluginUpdates?.length } ${ translate( 'Available' ) }`,
+				status: pluginUpdates?.length > 0 ? 'warning' : 'success',
+				type: 'plugin',
+				updates: pluginUpdates?.length,
+			};
+		},
+		[ translate ]
+	);
+};
+
+const formatErrorData = ( isConnected: boolean ): ErrorNode => ( {
+	status: isConnected ? 'success' : 'failed',
+	type: 'error',
+	value: isConnected ? 'error' : '',
+} );
+
+const useFormatSite = () => {
+	const formatBackupData = useFormatBackupData();
+	const formatMonitorData = useFormatMonitorData();
+	const formatPluginData = useFormatPluginData();
+	const formatScanData = useFormatScanData();
+
+	return useCallback(
+		( site: Site, isConnected: boolean ): SiteData => {
+			return {
+				site: {
+					value: site,
+					error: ! isConnected,
+					status: 'active',
+					type: 'site',
+				},
+				stats: formatStatsData( site ),
+				boost: formatBoostData( site ),
+				backup: formatBackupData( site ),
+				scan: formatScanData( site ),
+				monitor: formatMonitorData( site ),
+				plugin: formatPluginData( site ),
+				error: formatErrorData( isConnected ),
+				isFavorite: site.is_favorite,
+				isSelected: site.isSelected,
+				onSelect: site.onSelect,
+			};
+		},
+		[ formatBackupData, formatMonitorData, formatPluginData, formatScanData ]
+	);
+};
+
+type SiteConnectionStatus = {
+	ID: number;
+	connected: boolean;
+};
+
+/**
+ * Returns formatted sites
+ */
+const useFormattedSites = (
+	sites: Site[],
+	connectionStatus: SiteConnectionStatus[]
+): SiteData[] => {
+	const formatSite = useFormatSite();
+
+	return useMemo(
+		() =>
+			sites.map( ( site ) => {
+				const isConnected =
+					connectionStatus.find( ( status ) => status.ID === site.blog_id )?.connected || false;
```

Comment:
This file also copied from Jetpack manage to avoid conflicts. `isConnected` is the actual change here.